.travis.yml

```diff
@@ -1,19 +1,40 @@
 sudo: false
 
+language: haskell
+
+env:
+- USE_STACK=  GHC_VERSION=7.10.1 CABAL_VERSION=1.22
+- USE_STACK=1 GHC_VERSION=7.8.4  CABAL_VERSION=
+
 install:
 - mkdir -p ~/bin
 - curl -sL -o - https://github.com/commercialhaskell/stack/releases/download/v0.1.0.0/stack-0.1.0.0-x86_64-linux.gz | zcat > ~/bin/stack
 - chmod +x ~/bin/stack
-- export PATH=/opt/ghc/7.8.4/bin:$PATH
+- export PATH=/opt/ghc/$GHC_VERSION/bin:/opt/cabal/$CABAL_VERSION/bin:$PATH
 - export GHC_OPTIONS="-rtsopts -threaded"
 - export STACK="$HOME/bin/stack --system-ghc -j2"
 
 script:
-- $STACK build --ghc-options "$GHC_OPTIONS"
-- $STACK install --ghc-options "$GHC_OPTIONS"
-- cp ~/.local/bin/upcast .
-- strip -p --strip-unneeded --remove-section=.comment upcast
-- upx upcast
+- |
+   set -ex
+   if [ -n "$USE_STACK" ]
+   then
+     $STACK build --ghc-options "$GHC_OPTIONS"
+     $STACK install --ghc-options "$GHC_OPTIONS"
+     cp ~/.local/bin/upcast .
```

Comment:
But, Stack doesn't like a GHC version different than the one it knows about, so it doesn't really make sense to build GHC 7.10 with Stack. Or does it?